pennylane/transforms/sign_expand.py

```diff
@@ -0,0 +1,340 @@
+    norm = eigs[-1]
+    proj = np.identity(size, dtype="complex64")
+    proj += -2 * np.outer(np.conjugate(eigvecs[:, 0]), eigvecs[:, 0])
+    last_i = 1
+
+    dEs, mus, projs, times = [], [], [], []
+
+    for index in range(len(eigs) - 1):
+        dE = (eigs[index + 1] - eigs[index]) / 2
+        if np.isclose(dE, 0):
+            continue
+        dEs.append(dE)
+        mu = (eigs[index + 1] + eigs[index]) / 2
+        mus.append(mu)
+        time = np.pi / (2 * (norm + abs(mu)))
+        times.append(time)
+
+        for j in range(last_i, index + 1):
+            proj += -2 * np.outer(np.conjugate(eigvecs[:, j]), eigvecs[:, j])
+            last_i = index + 1
+
+        projs.append(proj.copy() * dE)
+
+    return dEs, mus, times, projs
+
+
+# pylint: disable=too-many-function-args)
+def construct_sgn_circuit(hamiltonian, tape, mus, times, phis, controls):
+    """
+    Takes a tape with state prep and ansatz and constructs the individual tapes
+    approximating/estimating the individual terms of your decomposition
+
+    Args:
+      hamiltonian (qml.Hamiltonian): The pennylane hamiltonian to be decomposed
+      tape (qml.QuantumTape: Tape containing the circuit to be expanded into the new circuits
+      mus (List[float]): The average between the two eigenvalues (E_1+E-2)/2
+      times (List[float]): The time for this term group to be evaluated/evolved at
+      phis (List[float]): Optimal phi values for the QSP part associated with the respective
+        delta and J
+      controls (List[control1, control2]): The two additional controls to implement the
+          Hadamard test and the quantum signal processing part on
+
+    Returns:
+      tapes (List[qml.tape]): Expanded tapes from the original tape that measures the terms
+        via the approximate sgn decomposition
+    """
+    coeffs = hamiltonian.data
+    tapes = []
+    for mu, time in zip(mus, times):
+        with tape.__class__() as new_tape:
+            # Put state prep and ansatz on tape in the 'old' register
+            for op in tape.operations:
+                op.queue()
+
+            # Put QSP and Hadamard test on the two ancillas Target and Control
+            qml.Hadamard(controls[0])
+            for i, phi in enumerate(phis):
+                qml.CRX(phi, wires=controls)
+                if i == len(phis) - 1:
+                    qml.CRY(np.pi, wires=controls)
+                else:
+                    evolve_under(hamiltonian.ops, coeffs, 2 * time, controls)
+                    qml.CRZ(-2 * mu * time, wires=controls)
+            qml.Hadamard(controls[0])
+
+            if tape.measurements[0].return_type == qml.measurements.Expectation:
+                qml.expval(-1 * qml.PauliZ(controls[0]))
+            else:
+                qml.var(qml.PauliZ(controls[0]))
+
+        tapes.append(new_tape)
+    return tapes
+
+
+@batch_transform
+def sign_expand(tape, circuit=False, J=10, delta=0.0, controls=['Hadamard','Target']):
```

Comment:
```suggestion
def sign_expand(tape, circuit=False, J=10, delta=0.0, controls=('Hadamard','Target')):
```
lists are mutable, therefor the default value might be mutated...